force/parser/parser.go

```diff
@@ -0,0 +1,52 @@
+package parser
+
+import (
+	"reflect"
+
+	"github.com/opendoor-labs/go-force/forcejson"
+	"github.com/pkg/errors"
+)
+
+func IsSlicePtr(i interface{}) (isptr bool) {
+	defer func() {
+		recover()
+	}()
+
+	slice := reflect.ValueOf(i).Elem()
+
+	// panic if slice doesn't behave like a slice
+	slice.Slice(0, 0)
+
+	return true
+}
+
+func ParseSFJSON(msg []byte, out interface{}) error {
+	err := forcejson.Unmarshal(msg, out)
+
+	if err == nil {
+		return nil
+	}
+
+	if !IsSlicePtr(out) {
+		return err
```

Comment:
suggestion: error should be more specific to this, i.e., it's neither a slice of salesforce objects nor a single salesforce object